File: java/base/src/main/java/vsdk/toolkit/raycasting/stochasticRaytracing/StochasticRayTracingState.java
/**
Options and runtime state for stochastic raytracing.
*/

package vsdk.toolkit.raycasting.stochasticRaytracing;

import vsdk.toolkit.render.ScreenBuffer;

public class StochasticRayTracingState {
    public StochasticRayTracingState() {
        samplesPerPixel = 1;
        progressiveTracing = 1;
        doFrameCoherent = 0;
        doCorrelatedSampling = 0;
        baseSeed = 0xFE062134L;
        radMode = RayTracingRadMode.STORED_NONE;
        nextEvent = 1;
        nextEventSamples = 1;
        lightMode = RayTracingLightMode.ALL_LIGHTS;
        backgroundDirect = 0;
        backgroundIndirect = 1;
        backgroundSampling = 0;
        scatterSamples = 1;
        differentFirstDG = 0;
        firstDGSamples = 36;
        separateSpecular = 0;
        reflectionSampling = RayTracingSamplingMode.BRDF_SAMPLING;
        minPathDepth = 5;
        maxPathDepth = 7;
        lastScreen = null;
    }

    // Pixel sampling
    public int samplesPerPixel;
    public int progressiveTracing;

    public int doFrameCoherent;
    public int doCorrelatedSampling;
    public long baseSeed;

    // Stored radiance handling
    public RayTracingRadMode radMode;

    // Direct Light sampling
    public int nextEvent;
    public int nextEventSamples;
    public RayTracingLightMode lightMode;

    // Background
    public int backgroundDirect;
    public int backgroundIndirect;
    public int backgroundSampling;

    // Scattering
    public int scatterSamples;
    public int differentFirstDG;
    public int firstDGSamples;
    public int separateSpecular;
    public RayTracingSamplingMode reflectionSampling;

    public int minPathDepth;
    public int maxPathDepth;

    public ScreenBuffer lastScreen;
}

File: typescript/base/src/main/vsdk/toolkit/raycasting/stochasticRaytracing/StochasticRayTracingState.ts
/**
Options and runtime state for stochastic raytracing.
*/

import { ScreenBuffer } from "../../render/ScreenBuffer";
import { RayTracingLightMode } from "./RayTracingLightMode";
import { RayTracingRadMode } from "./RayTracingRadMode";
import { RayTracingSamplingMode } from "./RayTracingSamplingMode";

export class StochasticRayTracingState {
  public samplesPerPixel: number;
  public progressiveTracing: number;

  public doFrameCoherent: number;
  public doCorrelatedSampling: number;
  public baseSeed: number;

  public radMode: RayTracingRadMode;

  public nextEvent: number;
  public nextEventSamples: number;
  public lightMode: RayTracingLightMode;

  public backgroundDirect: number;
  public backgroundIndirect: number;
  public backgroundSampling: number;

  public scatterSamples: number;
  public differentFirstDG: number;
  public firstDGSamples: number;
  public separateSpecular: number;
  public reflectionSampling: RayTracingSamplingMode;

  public minPathDepth: number;
  public maxPathDepth: number;

  public lastScreen: ScreenBuffer | null;

  public constructor() {
    this.samplesPerPixel = 1;
    this.progressiveTracing = 1;
    this.doFrameCoherent = 0;
    this.doCorrelatedSampling = 0;
    this.baseSeed = 0xFE062134;
    this.radMode = RayTracingRadMode.STORED_NONE;
    this.nextEvent = 1;
    this.nextEventSamples = 1;
    this.lightMode = RayTracingLightMode.ALL_LIGHTS;
    this.backgroundDirect = 0;
    this.backgroundIndirect = 1;
    this.backgroundSampling = 0;
    this.scatterSamples = 1;
    this.differentFirstDG = 0;
    this.firstDGSamples = 36;
    this.separateSpecular = 0;
    this.reflectionSampling = RayTracingSamplingMode.BRDF_SAMPLING;
    this.minPathDepth = 5;
    this.maxPathDepth = 7;
    this.lastScreen = null;
  }
}
